CRM/Mailing/BAO/MailingJob.php

```diff
@@ -450,6 +450,12 @@ public function queue($testParams = NULL) {
       $params = array();
       $count = 0;
       while ($recipients->fetch()) {
+        // CRM-18543: there are situations when both the email and phone are null.
+        // Skip the recipient in this case.
+        if (empty($recipients->email_id) && empty($recipients->phone_id)) {
+          continue;
+        }
+
         if ($recipients->phone_id) {
```

Comment:
The problem for me is not so much the 'null' strings (which are needed as part of the INSERT statement), but the disconnect between the calling code (which says that a mailing job is either 'sms' or 'email') and this code which allows individual recipients to be treated as either 'sms' or 'email'.